lampod/src/ln/channel_manager.rs

```diff
@@ -432,13 +432,19 @@ impl ChannelEvents for LampoChannelManager {
             }
         };
 
+        let txid = match &tx {
+            Some(t) => Some(t.txid()),
+            None => None,
+        };
+
         Ok(response::OpenChannel {
             node_id: open_channel.node_id,
             amount: open_channel.amount,
             public: open_channel.public,
             push_mst: 0,
             to_self_delay: 2016,
             tx,
+            txid,
         })
     }
```

Comment:
it is possible write this match in a single line?

A possible diff can be 

```diff
diff --git a/lampod/src/ln/channel_manager.rs b/lampod/src/ln/channel_manager.rs
index f65f67c..ab6f83d 100644
--- a/lampod/src/ln/channel_manager.rs
+++ b/lampod/src/ln/channel_manager.rs
@@ -432,11 +432,7 @@ impl ChannelEvents for LampoChannelManager {
             }
         };

-        let txid = match &tx {
-            Some(t) => Some(t.txid()),
-            None => None,
-        };
-
+        let txid = tx.as_ref().map(|tx| tx.txid());
         Ok(response::OpenChannel {
             node_id: open_channel.node_id,
             amount: open_channel.amount,
```

Reply:
Yep, your way is better. Updated.